File: components/events/TweetUncanceledEvent.ts
import { Environment, fetchQuery, graphql } from "relay-runtime"

import { createEventHook } from "components/EventsProvider"

export interface TweetUncanceledEvent {
  user_id: string
  tweet_id: string
}

const fetchTweetQuery = graphql`
  query TweetUncanceledEventQuery($id: ID!) {
    node(id: $id) {
      id
      ... on TweetGroup {
        status
      }
    }
  }
`

export async function handleTweetUncanceled(
  environment: Environment,
  event: TweetUncanceledEvent
): Promise<void> {
  await fetchQuery(environment, fetchTweetQuery, { id: event.tweet_id })
}

export const useTweetUncanceledEvent = createEventHook(
  "TweetUncanceled",
  handleTweetUncanceled
)

File: ios/Courier/Courier/Events/TweetUncanceledEvent.swift
import Combine
import Relay
import CourierGenerated

struct TweetUncanceledEvent: Decodable {
    var userID: String
    var itemID: String

    enum CodingKeys: String, CodingKey {
        case userID = "user_id"
        case itemID = "item_id"
    }
}

private let query = graphql("""
query TweetUncanceledEventQuery($id: ID!) {
  tweetGroup(id: $id) {
    id
    status
  }
}
""")

func handleTweetUncanceled(_ environment: Environment, _ event: TweetUncanceledEvent) -> AnyPublisher<Never, Error> {
    environment.fetchQuery(TweetUncanceledEventQuery(id: event.itemID))
        .ignoreOutput()
        .eraseToAnyPublisher()
}
